flamedisx/x1t_sr1.py

```diff
@@ -53,37 +162,61 @@ def add_extra_columns(self, d):
                           d['z'].values]))
 
     @staticmethod
-    def electron_detection_eff(drift_time, *, elife=650e3, extraction_eff=0.96):
+    def electron_detection_eff(drift_time, *, elife=def_elife, extraction_eff=def_extract_eff):
         #TODO: include function for elife time dependency
         return extraction_eff * tf.exp(-drift_time / elife)
 
     @staticmethod
-    def electron_gain_mean(s2_relative_ly, *, g2=11.4/(1.-0.63)/0.96):
+    def electron_gain_mean(s2_relative_ly, *, g2=def_g2/(1.-def_frac_top)/def_extract_eff):
         return g2 * s2_relative_ly
 
     @staticmethod
-    def electron_gain_std(s2_relative_ly, *, g2=11.4/(1.-0.63)/0.96):
-        return g2*0.96*0.25+0.*s2_relative_ly
+    def electron_gain_std(s2_relative_ly, *, g2=def_g2/(1.-def_frac_top)/def_extract_eff):
+        return g2*def_extract_eff*0.25+0.*s2_relative_ly
 
     #TODO: implement better the double_pe_fraction or photon_detection_efficiency as parameter
     @staticmethod
-    def photon_detection_eff(s1_relative_ly, g1 =0.142):
-        #g1 = 0.142 from paper
-        mean_eff= g1 / (1. + 0.219)
+    def photon_detection_eff(s1_relative_ly, g1 =def_g1):
+        mean_eff= g1 / (1. + def_p_dpe)
         return mean_eff * s1_relative_ly
 
+    def s1_acceptance(self, s1, photon_detection_eff, photon_gain_mean, mean_eff=def_g1 / (1 + def_p_dpe), 
+            cs1_min=3, cs1_max=70):
+        print('s1_acceptance: cs1 min = %i, cs1 max = %f' % (cs1_min, cs1_max))
+        cs1 = mean_eff * s1 / (photon_detection_eff * photon_gain_mean)
+        mask = tf.where((cs1 > cs1_min) & (cs1 < cs1_max),
+                        tf.ones_like(s1, dtype=fd.float_type()),
+                        tf.zeros_like(s1, dtype=fd.float_type()))
+
+        # multiplying by combined cut acceptance
+        cut_accept_wgt = itp_cut_accept_tf(s1, self.cut_accept_map_s1, self.cut_accept_support_s1)
+        mask_out = tf.math.multiply(mask, cut_accept_wgt)
 
+        return mask_out
+
+    def s2_acceptance(self, s2, electron_detection_eff, electron_gain_mean,
+        cs2b_min=50.1, cs2b_max=7940):
+        print('s2_acceptance: cs2b min = %i, cs2b max = %f' % (cs2b_min, cs2b_max))
+        cs2 = (def_g2/def_extract_eff) * s2 / (electron_detection_eff*electron_gain_mean)
+        mask = tf.where((cs2 > cs2b_min) & (cs2 < cs2b_max),
+                        tf.ones_like(s2, dtype=fd.float_type()),
+                        tf.zeros_like(s2, dtype=fd.float_type()))
+
+        # multiplying by combined cut acceptance
+        cut_accept_wgt = itp_cut_accept_tf(s2, self.cut_accept_map_s2, self.cut_accept_support_s2)
+        mask_out = tf.math.multiply(mask, cut_accept_wgt)
+
+        return mask_out
 
 # ER Source for SR1
 @export
 class SR1ERSource(SR1Source,fd.ERSource):
 
     @staticmethod
-    def p_electron(nq, W=13.8e-3, mean_nexni=0.135,  q0=1.13, q1=0.47,
+    def p_electron(nq, W=13.8e-3, mean_nexni=0.15,  q0=1.13, q1=0.47,
```

Comment:
Could you comment in the PR why you changed the values? (I think you did it for a good reason, but best to have a record of why somewhere)

Reply:
Changed value of `mean_nexni` in `p_electron` in `SR1ERSource` in `x1t_sr1.py` from 0.135 to 0.15 in accordance to value in table II of analysis paper II (1902.11297v2).

Updated PR to include this.